keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -151,3 +169,168 @@ def get_config(self):
             }
         )
         return config
+
+    def augment_bounding_boxes(self, bounding_boxes, transformation, **kwargs):
+        image = None
+        image = kwargs.get("image")
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomShear()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomShear(bounding_box_format='xyxy')`"
+            )
+        height, width, _ = image.shape
+
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+            dtype=self.compute_dtype,
+        )
+        x, y = transformation
+        # apply horizontal shear
+        if x is not None:
+            bounding_boxes = self._apply_horizontal_transformation_to_bounding_box(
```

Comment:
Can we convert to extended format, perform both shears, then convert back to standard bounding box formats at the end?  This will actually give better accuracy.

Reply:
Can be done. But it will introduce more complexity when converting from 8 coordinate system to 4 coordinate system.

Reply:
Done